### PR TITLE
Fix no copy when cast `Snippet` to `DisplayList` and #29

### PR DESCRIPTION
**Pull request by botika**
With this pull request the "no copy" process is completed. All the crate is free copy :smile: 
I have changed from `yaml` to `toml` because `yaml` does not implement `serde::Deserialize`

### CODE REVIEW

**Comment by botika** [comment]
Fixed #29

File: src/display_list/from_snippet.rs [hunk 1]
```diff
@@ -2,6 +2,52 @@
...
+        CursorLines(src)
+    }
+}
+
+#[repr(u8)]
+enum EndLine {
+    EoF = 0,
+    One = 1,
+    Two = 2,
+}
```
Comment (zbraniecki) [inline src/display_list/from_snippet.rs:18]: Why do you use `repr(u8)` on this?
Also, can you name it CRLF and LF respectively? EOF is also all capitals.

File: src/display_list/from_snippet.rs [hunk 2]
```diff
@@ -236,9 +282,7 @@ fn format_body(slice: snippet::Slice<'_>, has_footer: bool) -> Vec<DisplayLine<'
     let mut current_index = 0;
     let mut line_index_ranges = vec![];
 
-    let lines = slice.source.lines();
-    let lines_len = lines.clone().count();
-    for (i, line) in lines.enumerate() {
+    for (line, end_line) in CursorLines::new(slice.source) {
```
Comment (zbraniecki) [inline src/display_list/from_snippet.rs:285]: I really like this solution!

File: src/display_list/from_snippet.rs (hunk 1 above)
Comment (botika) [reply src/display_list/from_snippet.rs:17]: It's done

**Comment by zbraniecki** [comment]
Looks great! Thank you!

**Pull request merged**